### DISCUSSION TITLE
Create Custom Brush similer to freedrawing brush with line smoothing applied

### DISCUSSION THREAD

**Discussion by Exlord** [fabricjs/fabric.js#7118] [Q&A]
Hi, I created a custom brush. But the lines and corners are sharp. In the freedrawing demo after a line is drawn and after mouseup the some transformation is applied to the line that makes it look smoother!

I browsed throw the source code and even used debugged in browser to see what exactly is happening but could not figure it out!

The only parts that seems to be related is `makePathSimpler` in `path` and `decimate` in `brush` .

Could you please point me in the write direction ?

**Comment by asturur** [comment]
Hi @Exlord if you can share your custom brush we can see what is it doing and why isn't sharp.
